prost-types/src/datetime.rs

```diff
@@ -730,15 +732,16 @@ mod tests {
         // Leap day
         assert_eq!(
             "2020-02-29T01:02:03.00Z".parse::<Timestamp>().unwrap(),
-            Timestamp::from(DateTime {
+            Timestamp::try_from(DateTime {
                 year: 2020,
                 month: 2,
                 day: 29,
                 hour: 1,
                 minute: 2,
                 second: 3,
                 nanos: 0,
-            }),
+            })
+            .unwrap(),
```

Comment:
ah maybe we can't, since these have different error types. would it be sensible to use `TimestampError` for the `TryFrom`?

Reply:
I think it is a bit lame that it can only return the one enum variant, but I agree that it is sensible. And it is an internal conversion, so we can change it if that makes sense at a later time.